http/options_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package http
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestOptions_Default ensures that the default values are as expected.
+func TestOptions_Default(t *testing.T) {
+	opts := getDefaultOptions()
+	require.NotNil(t, opts)
+	require.Equal(t, "", opts.withRedactionValue)
+}
+
+// TestOptions_WithRedactionValue ensures that we set the correct value to use for
+// redaction when required.
+func TestOptions_WithRedactionValue(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		Value           string
+		ExpectedValue   string
+		IsErrorExpected bool
+	}{
+		"empty": {
+			Value:           "",
+			ExpectedValue:   "",
+			IsErrorExpected: false,
+		},
+		"whitespace": {
+			Value:           "     ",
+			ExpectedValue:   "     ",
+			IsErrorExpected: false,
+		},
+		"value": {
+			Value:           "*****",
+			ExpectedValue:   "*****",
+			IsErrorExpected: false,
+		},
+	}
+
+	for name, tc := range tests {
+		name := name
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			opts := &listenerConfigOptions{}
+			applyOption := WithRedactionValue(tc.Value)
+			err := applyOption(opts)
+			switch {
+			case tc.IsErrorExpected:
+				require.Error(t, err)
+			default:
+				require.NoError(t, err)
+				require.Equal(t, tc.ExpectedValue, opts.withRedactionValue)
+			}
+		})
+	}
+}
+
+// TestOptions_WithRedactAddresses ensures that the option works as intended.
+func TestOptions_WithRedactAddresses(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		Value           bool
+		ExpectedValue   bool
+		IsErrorExpected bool
+	}{
+		"true": {
+			Value:           true,
+			ExpectedValue:   true,
+			IsErrorExpected: false,
```

Comment:
❓ There are not any tests where `IsErrorExpected` is `true`. Are there any cases that we could test where an error would be expected? If not, would it make sense to remove it?

Reply:
Ah yes, thanks Chris that's a good point! I'll pull out the field as we should just assert there's never an error and that's enough.